### DISCUSSION TITLE
Does tart really copy 50 gigabytes every time tart clone is run?

### DISCUSSION THREAD

**Discussion by sschultze** [cirruslabs/tart#292] [General]
I want to start with a fresh VM every time, so I'm calling

```
tart clone ghcr.io/cirruslabs/macos-ventura-base:latest ventura-base
```

before doing my work. (This seems to also delete any existing VM with this name first, which is good.) Surprisingly, running `ventura clone` only takes a small fraction of a second, even though a 50 GB subfolder is created in the `~/.tart/vms` folder.

Does `tart clone` or macOS really copy all the 50 GB within the SSD, putting it as risk to wear out quite fast over time if I'm running the command too often? Or is there some sector-based deduplication taking place in either tart or in the operating system itself?

Thank you in advance for clarifying.

**Comment by fkorotkov** [comment]
There is no risk of wear out your SSD. All the magic is happening thanks to [APFS Copy-on-Write](https://mac-optimization.bestreviews.net/the-magic-behind-apfs-copy-on-write/). In the nutshell `tart copy` is instant because there is no physical coping is happing and as you run your cloned VM the disk is changing and only the changes are getting written to the physical SSD.

**Reply by sschultze** [reply]
This is great, thank you! :-) And also thanks for the really quick response!
